judge/utils/judge_api.py

```diff
@@ -0,0 +1,172 @@
+import re
+
+import requests
+from django.conf import settings
+from django.core.cache import cache
+
+from judge.models import Problem
+
+
+class APIError(Exception):
+    pass
+
+
+class OJAPI:
+    @staticmethod
+    def get_problem_data(url):
+        """Return codename and judge name (for choosing API)."""
+        presets = settings.OJ_PROBLEM_PRESET
+
+        for preset in presets:
+            regex = re.compile(preset['regex']).match(url)
+            if regex is None:
+                continue
+            codename = preset['codename'] % regex.groups()
+            return {
+                'codename': codename,
+                'judge': preset['judge'],
+            }
+
+        raise APIError('This link or online judge is not currently supported.')
+
+    @staticmethod
+    def CodeforcesProblemAPI(codename):
+        contestset = cache.get('OJAPI_data_Codeforces', None)
+
+        if contestset is None:
```

Comment:
or `contest_id` > `max contest_id` in the `contestset`. Because there might be new contest coming, and we need to fetch that.